### DISCUSSION TITLE
How do I index additional fields?

### DISCUSSION THREAD

**Discussion by vrdn-23** [castorini/pyserini#1266] [Q&A]
I can see from the documentation that we can potentially incorporate additional fields like NER information. What I want to know: 
1) Is there a particular setting that I have to turn on while indexing to make sure that additional fields get indexed? Or is the only way to index them using the --storeRaw setting?
2) When I follow the command provided in the main README to index your own documents, I see the returned "hit" results have a docid attribute but not a content attribute. Does that mean that the returned result is based on the score from the entire raw document or just the contents part?
3) Can I have a sparse retrieval over multiple fields? Or specific ones? Like for example only over "contents" and another field "title"?
4) Is there another website or documentation page other than the documentation on Github?

Thanks for the amazing work on the project. Looking forward to hear your response!

**Comment by GioiaZheng** [comment]
A few separate things are getting mixed up here:

1. **`--storeRaw` vs searchable fields are different.** `--storeRaw` just keeps the original JSON so you can read it back at search time; it does **not** make extra keys searchable. To index something like NER or title as its own searchable field, put it as a top-level key in your JSONL and pass `--fields title ner` at index time (Anserini multifield indexing). Use `--storeContents` if you also want the analyzed text returned.

2. **docid but no content** is expected. Search returns docids + scores only. To get text back you must have indexed with `--storeRaw`/`--storeContents`, then fetch it:

```python
hits = searcher.search('query')
raw = searcher.doc(hits[0].docid).raw()       # original JSON
contents = searcher.doc(hits[0].docid).contents()
```

Scoring uses the analyzed `contents` field, not the stored raw.

3. **Multi-field retrieval:** yes — index with `--fields`, then weight fields at query time:

```python
searcher.search('query', fields={'contents': 1.0, 'title': 0.5})
```

4. **Docs:** the Pyserini `docs/usage-index.md` and `docs/usage-search.md` in the repo, plus the Anserini `IndexCollection` options for the `-fields` / `-storeRaw` / `-storeContents` semantics.
